cscs-checks/prgenv/cuda/cuda_aware_mpi.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.simple_test
+class CudaAwareMPICheck(rfm.CompileOnlyRegressionTest):
+    def __init__(self):
+        super().__init__()
+        self.descr = 'Cuda-aware MPI test from the NVIDIA repo.'
+        self.sourcesdir = ('https://github.com/NVIDIA-developer-blog/'
+                           'code-samples.git')
+        self.valid_systems = ['daint:gpu', 'dom:gpu',
+                              'kesch:cn', 'tiger:gpu',
+                              'arolla:cn', 'tsa:cn',
+                              'ault:amdv100', 'ault:intelv100']
+        if self.current_system.name == 'kesch':
```

Comment:
Does it make sense to parameterize this test and pass all these combinations as parameters?

Reply:
I see what you mean, but this seems to be the way things are implemented in many other cases. From what I understand, parametrised tests are used to change actual test parameters, not stuff related to the environments. Perhaps this could just go in the env settings?

Reply:
Generally, we can parametrize tests on anything and there are cases that it is useful to parametrize even on valid systems and environments. In this case, I honestly don't what would be the best solution :-)